### PR TITLE
Only allow declares inside declare module

### PR DESCRIPTION
**Pull request by danez**
This is the followup of #71 

This now checks for the term `declare` being used inside `declare module`. Previously everything was allowed instead of declare. https://astexplorer.net/#/Y6hVjJkQ2m

### CODE REVIEW

File: test/fixtures/flow/declare-module/8/options.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "throws": "Unexpected token (2:2)"
```
Comment (hzoo) [inline test/fixtures/flow/declare-module/8/options.json]: Is it possible to have a better error message here?

Comment (danez) [inline test/fixtures/flow/declare-module/8/options.json]: Maybe something like: "Only declare statements are allowed inside declare module {}"

Comment (hzoo) [inline test/fixtures/flow/declare-module/8/options.json]: Yeah something like that. Could even give an example (we haven't really done that)

I've been liking elm's errors/messages http://elm-lang.org/blog/compilers-as-assistants

**Comment by danez** [comment]
I changed the error message to be a little bit nicer.

File: src/parser/util.js [hunk 2]
```diff
@@ -79,6 +79,6 @@ pp.expect = function (type) {
 
 // Raise an unexpected token error.
 
-pp.unexpected = function (pos) {
-  this.raise(pos != null ? pos : this.state.start, "Unexpected token");
+pp.unexpected = function (pos, message = "Unexpected token") {
+  this.raise(pos != null ? pos : this.state.start, message);
```
Comment (hzoo) [inline src/parser/util.js]: I like being able to change the error message. Unexpected token is not fun to see (my example is `function a() { await b; }`) without async in front of function

**Comment by hzoo** [comment]
k 👍 , @jeffmo

**Comment by gabelevi** [comment]
👍 this looks correct to me. Everything inside of a `declare module` declaration does start with `declare`.

**Pull request merged**